ts/backend/src/app/features/controller/controller.spec.ts

```diff
@@ -0,0 +1,106 @@
+import type { Express } from 'express'
+import express from 'express'
+import supertest from 'supertest'
+import TestAgent from 'supertest/lib/agent'
+import { beforeAll, describe, expect, test } from 'vitest'
+import { configuration } from '../config/config.mjs'
+import { defaults } from '../config/defaults.mjs'
+import { Controller } from './controller.mjs'
+
+// Custom test controller to test Controller's basic functionality.
+// All `attach<Verb>` methods are aliases for the private `attach`. Hence let
+// us assume that by testing one (i.e. `attachGet`), all of them are tested
+// sufficiently.
+class TestController extends Controller {
```

Comment:
To avoid confusion that your unit under test is `Controller`, consider instantiating `Controller` with the necessary args or patch it.